platform/commonUI/browse/src/ObjectHeaderController.js

```diff
@@ -41,25 +41,44 @@ define(
          * @param event the mouse event
          */
         ObjectHeaderController.prototype.updateName = function (event) {
-            if (event && (event.type === 'blur' || event.which === 13)) {
-                var name = event.currentTarget.innerHTML;
+            if (!event) {
+                return;
+            }
+
+            if (this.enterKeyHandled === true) {
+                this.enterKeyHandled = false;
+                return;
+            }
+
+            if (event.type === 'blur') {
+                this.updateModel(event);
+            } else if (event.which === 13) {
+                this.updateModel(event);
```

Comment:
Seems like `this.updateModel(event)` could just be moved out of the `if` block?

Reply:
Disregard; moving this out of the `if` block would trigger updates on every `keypress`, which is not what we went